pyrasterframes/python/README.md

```diff
@@ -0,0 +1,101 @@
+# PyRasterFrames
+
+PyRasterFrames is the Python API for Spark RasterFrames. 
+
+Building RasterFrames requires [`sbt`](https://www.scala-sbt.org/), and all `sbt` commands referenced below must be 
+run from the root source directory, i.e. the parent of the `pyrasterframes` directory. Know that `sbt` is much 
+faster if run in "interactive" mode, where you launch `sbt` with no arguments, and subsequent commands are invoked
+via an interactive shell. But for context clarity, we'll prefix each command example below with `sbt`. 
+
+## Quickstart
+
+The quickest way to run a `pyspark` shell with the latest RasterFrames enabled is to run:
+
+```bash
+sbt pySparkCmd
+```
+
+This will: 
+
+1. Compile all the Scala/JVM code.
+1. Merge all JVM code and dependencies into a single "assembly" JAR file.
+1. Create the PyRasterFrames `.whl` package
+1. Construct a temporary initialization script
+1. Emit a `bash` command with requiesite arguments to load RasterFrames and import PyRasterFrames 
+
+You then copy/paste the emitted command into your `bash` shell to start up a spark shell. It assumes you have
+`pyspark` >= 2.3.2 installed in your environment.
+
+## Running Tests and Examples
+
+The PyRasterFrames unit tests can found in `<src-root>/pyrasterframes/python/tests`. To run them:
+
+```bash
+sbt pyTests
+```
+
+Similarly, to run the examples in `<src-root>pyrasterframes/python/examples`:
+
+```bash
+sbt pyExamples
+```
+
+## Creating and Using a Build
+
+Assuming that `$SCALA_VER` is the major verison of Scala in use, and `$VER` is the version of RasterFrames, 
+the primary build artifacts are:
+
+* JVM library: `pyrasterframes/target/scala-$SCALA_VER/pyrasterframes_$SCALA_VER-$VER.jar`
+* Python package: `pyrasterframes/target/scala-$SCALA_VER/pyrasterframes-python-$VER.zip`
+
+You build them with"
+
+```bash
+sbt pyrasterframes/package
```

Comment:
Of interest here the JAR artifact is

    pyrasterframes/target/scala-2.11/pyrasterframes_2.11-0.8.0-SNAPSHOT.jar

But the one from `sbt pySparkCmd` is 

    pyrasterframes/target/scala-2.11/pyrasterframes-assembly-0.8.0-SNAPSHOT.jar 

Is it possible to normalize those? Or are the contents of the jars intended to differ?

Reply:
They are two different things. The first one is what you publish to Maven central, and all that's necessary for programs that wish to include the JVM code, and where a build tool will automatically load all the transitive dependencies. The second is the assembly containing everything, and isn't something you would ever publish to Maven central (not allowed I think, but certainly frowned upon). 

IMO, assemblies are evil, but I've stopped fighting the impulse people who don't understand the Maven ecosystem have to use them. It seems to come with the territory, especially in the Spark world.